whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapService.java

```diff
@@ -486,6 +529,34 @@ private String objectTypesToString(final Collection<ObjectType> objectTypes) {
         return COMMA_JOINER.join(objectTypes.stream().map(ObjectType::getName).toList());
     }
 
+    private List<RpslObject> handleRelationQuery(final HttpServletRequest request, final RdapRequestType requestType,
+                                                 final RelationType relationType, final String key) {
```

Comment:
Do you think it makes more sense to have a separate  controller for rirsearch ?

Reply:
I don't think so because there is always one endpoint and it is part of RDAP

Reply:
Thanks it's just that this class is getting so big, infact we pu so many logic in controller level but I agree it is been like this forever